content/post/practical-stegen.Rmd

```diff
@@ -0,0 +1,861 @@
+new_labels # check the result
+names(stegen) <- new_labels
+```
+
+We set convert the unique identifiers to character strings (`character`), dates
+of onset to actual `Date` objects, and sex and illness are set to categorical variables (`factor`):
+
+```{r stegen_unique_ids}
+stegen$unique_key <- as.character(stegen$unique_key)
+stegen$sex <- factor(stegen$sex)
+stegen$ill <- factor(stegen$ill)
+stegen$date_onset <- as.Date(stegen$date_onset)
+```
+
+We use the function `recode()` from the *dplyr* package to recode `sex` more
+explicitely:
+
+```{r stegen_recode}
+stegen$sex <- recode_factor(stegen$sex, "0" = "male", "1" = "female")
+stegen$ill <- recode_factor(stegen$ill, "0" = "non case", "1" = "case")
+```
+
+
+Finally we look in more depth into these variables having maximum values of 9,
+where we expect 0/1; `table()` is useful to list all values taken by a variable,
+and listing their frequencies:
+
+```{r stegen_nines}
+
+table(stegen$pork)
+table(stegen$salmon)
+table(stegen$horseradish)
+
+```
+
+The only rogue values are `9`; they are likely either data entry issues, or
+missing data, which in **R** should be coded as `NA` ("not available"). We can
+replace these values using:
+
+```{r stegen_replace_9}
+stegen$pork[stegen$pork == 9] <- NA
+stegen$salmon[stegen$salmon == 9] <- NA
+stegen$horseradish[stegen$horseradish == 9] <- NA
+```
+
+
+<details>
+
+<summary> <b> Going further </b> Why does this work? Click here for more details. </summary>
+
+
+There are several things going on in a command like:
+
+```{r eval = FALSE}
+stegen$pork[stegen$pork == 9] <- NA
+```
+
+let us break them down:
+
+1. `stegen$pork` means "get the variable called `pork` in the dataset `stegen`"
+2. `[...]` subset the vector according to `...`; if `...` is a series of
+   `TRUE/FALSE` values, only entries corresponding to `TRUE` are retained
+3. `... == 9` for each value of `...`  test if it is equal to `9`, and return `TRUE` if so (`FALSE` otherwise) 
+4. `... <- NA` replace `...` with `NA` (missing value)
+
+in other words: "isolate the entries of `stegen$pork` which equal `9`, and
+replace them with `NA`; here is another toy example to illustrate the procedure:
+
+```{r stegen_subset}
+## make toy input vector
+toy_vector <- 1:5
+toy_vector
+
+## make toy logical vector for subsetting
+toy_logical <- c(FALSE, TRUE, TRUE, FALSE, TRUE)
+toy_logical
+toy_vector[toy_logical] # subset values
+toy_vector[toy_logical] <- 0 # replace subset values
+toy_vector # check outcome
+
+```
+
+</details>
+
+## Saving clean data
+
+Now that we've cleaned our data, we will probably want to re-use it in further
+analyses, but we probably don't want to go through the steps of cleaning it
+again. Here, we will save our cleaned data into a new folder under `data/`
+called `cleaned/`. We will use **R**'s function `dir.create()` to create this
+folder under our data folder:
+
+```{r create-clean-dir, eval = FALSE}
+clean_dir <- here("data", "cleaned")
+dir.create(clean_dir)
```

Comment:
here too - would tell them to go look in the folder and see that a new one has been created!